spec/request/request_ip_spec.rb

```diff
@@ -0,0 +1,163 @@
+    # Reset the Conjur routes to remove the test endpoint
+    Rails.application.reload_routes!
+  end
+
+  # --------------------------------------------------------------------
+  # Test Setup
+  # --------------------------------------------------------------------
+
+  let(:body_hash) { JSON.parse(response.body) }
+
+  # We can't modify the access token middleware to add an exception to our
+  # test route, so we need to create an access token for our request to use.
+  let(:bearer_token) { Slosilo["authn:rspec"].signed_token('admin') }
+  let(:token_auth_header) do
+    "Token token=\"#{Base64.strict_encode64 bearer_token.to_json}\""
+  end
+
+  let(:request_env) do
+    { 'HTTP_AUTHORIZATION' => token_auth_header, 'REMOTE_ADDR' => remote_ip }
+  end
+
+  let(:headers) do
+    {}.tap do |h|
+      h['X-Forwarded-For'] = x_forwarded_for if x_forwarded_for
+    end
+  end
+
+  # The value to user for the TRUSTED_PROXIES environment variable
+  let(:trusted_proxies) { nil }
+
+  # The value to use for the X-Forwarded-For HTTP header
+  let(:x_forwarded_for) { nil }
+
+  # --------------------------------------------------------------------
+  # Test Scenarios
+  # --------------------------------------------------------------------
+
+  # Without any other configuration in play, we expect to get the remote
+  # TCP connection IP address as the request IP address.
+  let(:remote_ip) { '44.0.0.1' }
+  let(:expected_request_ip) { '44.0.0.1' }
+  include_examples "expected request IP address"
+
+  # When X-Forwarded-For is set from an untrusted remote IP, we ignore it
+  context 'when X-Forwarded-For is set' do
+    let(:x_forwarded_for) { '3.3.3.3' }
+    let(:expected_request_ip) { '44.0.0.1' }
+    include_examples "expected request IP address"
+
+    # Loopback is a default trusted IP, so we accept the forwarded address
+    context 'when remote IP is loopback' do
+      let(:remote_ip) { '127.0.0.1' }
+      let(:expected_request_ip) { '3.3.3.3' }
+      include_examples "expected request IP address"
+    end
+  end
+
+  context 'when TRUSTED_PROXIES is set' do
+    # If TRUSTED proxies are set, and are remote IP address is not trusted
+    # we expect to get the remote IP
+    let(:trusted_proxies) { '4.4.4.4,5.5.5.5' }
+    let(:expected_request_ip) { '44.0.0.1' }
+    include_examples "expected request IP address"
+
+    context 'when remote IP is loopback' do
+      let(:remote_ip) { '127.0.0.1' }
+      let(:expected_request_ip) { '127.0.0.1' }
+      include_examples "expected request IP address"
+
+      # Loopback is always trusted, even if not explicity provided in
+      # TRUSTED_PROXIES, so we accept the forwarded IP address.
+      context 'when X-Forwarded-For is set' do
+        let(:x_forwarded_for) { '3.3.3.3' }
+        let(:expected_request_ip) { '3.3.3.3' }
+        include_examples "expected request IP address"
+      end
+    end
+
+    # The following spec exercise how variations on the X-Forwarded-For header
```

Comment:
Hi,
I don't see a test that simulate our issue of having duplicate headers.
Currently we get from LB `X-Forwarded-For` but nginx is adding `X-Forwarded-For`
Need to either [fix this issue in nginx](https://github.com/conjurinc/appliance/pull/1300) or test for it so we can better simulate the production use

Reply:
@Tovli, I am not trying to test for that yet, nor address that in this PR.

This PR is to resolve a regression found in Conjur from Rails 5 while work on the Client IP Sourcing doc here: https://github.com/cyberark/dap-wiki/pull/85.

I haven't gotten to the DAP portion in that document yet, and I expect this to come up then.

As a side note, I do agree that the issue should probably be fixed in DAP's nginx reverse proxy though.